modules/contest-2021/ContestRegister.tsx

```diff
@@ -0,0 +1,771 @@
+  );
+
+  const handleTechAccessChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setTechAccess({
+      ...techAccess,
+      [event.target.name]: event.target.checked,
+    });
+  };
+
+  const getTechAccessList = (): string => {
+    let list = '';
+    Object.entries(techAccess).forEach(([tech, access]) => {
+      if (access) {
+        list += `${list.length ? ', ' : ''}${tech}`;
+      }
+    });
+    return list;
+  };
+
+  const onSubmit = (data: FormData) => {
+    setSubmitSuccess(false);
+    setSubmitLoading(true);
+    setDuplicateParentWarning(false);
+    setSubmitError(undefined);
+
+    const participantData = {
+      ...data,
+      DOB: moment(data.DOB).format(dateFormat),
+      TECHACCESS: getTechAccessList(),
+      ISPARTIC: 'true',
+    };
+    const parentData = {
+      FNAME: data.P_FNAME || '',
+      LNAME: data.P_LNAME || '',
+      EMAIL: data.P_EMAIL || '',
+      PARENT: 'true',
+    };
+
+    console.log(participantData);
+    console.log(parentData);
+
+    dispatch(subscribeToList(participantData, 'participant'));
+    dispatch(subscribeToList(parentData, 'parent'));
+  };
+
+  useEffect(
+    () => {
+      if (mailchimpState.error) {
+        setSubmitError(mailchimpState.error);
+        setSubmitLoading(false);
+      }
+      else {
+        if (
+          !mailchimpState.loading
+          && mailchimpState.submissions
+          && mailchimpState.submissions.length
+        ) {
+          const participantIndex = mailchimpState.submissions.findIndex(
+            (sub => sub.type === 'participant')
+          );
+          const parentIndex = mailchimpState.submissions.findIndex(
+            (sub => sub.type === 'parent')
+          );
+
+          console.log(participantIndex, parentIndex);
+
+          if (participantIndex >= 0 && parentIndex >= 0) {
+            if (mailchimpState.submissions[participantIndex].code === 'error') {
+              const registeredError =
+                'Oops! According to our records you\'ve already registered for Digital Waves 2021. '
+                + 'If you have not registered already and would like to register with this email '
+                + 'please contact info@digitalwavesnl.ca and we will assist you.';
+
+              setSubmitError(new Error(registeredError));
+              setSubmitLoading(false);
+            }
+            else {
+              if (mailchimpState.submissions[parentIndex].code === 'error') {
+                setDuplicateParentWarning(true);
+              }
+              setSubmitSuccess(true);
+              setSubmitLoading(false);
+            }
+          }
+        }
+        if (
+          submitLoading
+          && !mailchimpState.loading
+          && mailchimpState.submissions
+          && !mailchimpState.submissions.length
+        ) {
+          setSubmitError(new Error(defaultErrorMsg));
+          setSubmitLoading(false);
+        }
+      }
+    },
+    [mailchimpState]
```

Comment:
I could be wrong, but I think `submitLoading` should be per the eslint exhaustive dep rule. When working with react hooks I recommend this eslint library, it's created by the developers of react: https://www.npmjs.com/package/eslint-plugin-react-hooks

Reply:
Good catch - wanna add this in a future PR?